File: internal/model/group_routing.go
package model

type GroupRoutingPreview struct {
	GroupID            int                     `json:"group_id"`
	GroupName          string                  `json:"group_name"`
	GroupMode          GroupMode               `json:"group_mode"`
	HealthScoreEnabled bool                    `json:"health_score_enabled"`
	Candidates         []GroupRoutingCandidate `json:"candidates"`
}

type GroupRoutingCandidate struct {
	GroupItemID         int      `json:"group_item_id"`
	Rank                int      `json:"rank"`
	ChannelID           int      `json:"channel_id"`
	ChannelName         string   `json:"channel_name"`
	ChannelKeyID        int      `json:"channel_key_id,omitempty"`
	SiteID              int      `json:"site_id,omitempty"`
	SiteName            string   `json:"site_name,omitempty"`
	SiteAccountID       int      `json:"site_account_id,omitempty"`
	SiteAccountName     string   `json:"site_account_name,omitempty"`
	ModelName           string   `json:"model_name"`
	Priority            int      `json:"priority"`
	Weight              int      `json:"weight"`
	Enabled             bool     `json:"enabled"`
	HealthScore         float64  `json:"health_score"`
	SampleCount         int      `json:"sample_count"`
	SuccessCount        int      `json:"success_count"`
	FailureCount        int      `json:"failure_count"`
	SuccessRate         float64  `json:"success_rate"`
	EmptyResponseRate   float64  `json:"empty_response_rate"`
	RateLimitCount      int      `json:"rate_limit_count"`
	AvgTTFBMS           int      `json:"avg_ttfb_ms"`
	AvgTotalMS          int      `json:"avg_total_ms"`
	CoolingDown         bool     `json:"cooling_down"`
	CooldownRemainingMS int64    `json:"cooldown_remaining_ms"`
	CooldownReason      string   `json:"cooldown_reason,omitempty"`
	QuotaStatus         string   `json:"quota_status"`
	QuotaBalance        float64  `json:"quota_balance,omitempty"`
	QuotaUsed           float64  `json:"quota_used,omitempty"`
	EffectiveScore      float64  `json:"effective_score"`
	Decision            string   `json:"decision"`
	Notes               []string `json:"notes,omitempty"`
}

File: web/src/api/endpoints/group-routing.ts
import { useQuery } from '@tanstack/react-query';
import { apiClient } from '../client';
import type { GroupMode } from './group';

export interface GroupRoutingCandidate {
    group_item_id: number;
    rank: number;
    channel_id: number;
    channel_name: string;
    channel_key_id?: number;
    site_id?: number;
    site_name?: string;
    site_account_id?: number;
    site_account_name?: string;
    model_name: string;
    priority: number;
    weight: number;
    enabled: boolean;
    health_score: number;
    sample_count: number;
    success_count: number;
    failure_count: number;
    success_rate: number;
    empty_response_rate: number;
    rate_limit_count: number;
    avg_ttfb_ms: number;
    avg_total_ms: number;
    cooling_down: boolean;
    cooldown_remaining_ms: number;
    cooldown_reason?: string;
    quota_status: string;
    quota_balance?: number;
    quota_used?: number;
    effective_score: number;
    decision: string;
    notes?: string[];
}

export interface GroupRoutingPreview {
    group_id: number;
    group_name: string;
    group_mode: GroupMode;
    health_score_enabled: boolean;
    candidates: GroupRoutingCandidate[];
}

function normalizeCandidate(candidate: Partial<GroupRoutingCandidate>): GroupRoutingCandidate {
    return {
        group_item_id: typeof candidate.group_item_id === 'number' ? candidate.group_item_id : 0,
        rank: typeof candidate.rank === 'number' ? candidate.rank : 0,
        channel_id: typeof candidate.channel_id === 'number' ? candidate.channel_id : 0,
        channel_name: candidate.channel_name ?? '',
        channel_key_id: typeof candidate.channel_key_id === 'number' ? candidate.channel_key_id : undefined,
        site_id: typeof candidate.site_id === 'number' ? candidate.site_id : undefined,
        site_name: candidate.site_name ?? '',
        site_account_id: typeof candidate.site_account_id === 'number' ? candidate.site_account_id : undefined,
        site_account_name: candidate.site_account_name ?? '',
        model_name: candidate.model_name ?? '',
        priority: typeof candidate.priority === 'number' ? candidate.priority : 0,
        weight: typeof candidate.weight === 'number' ? candidate.weight : 1,
        enabled: candidate.enabled !== false,
        health_score: typeof candidate.health_score === 'number' ? candidate.health_score : 100,
        sample_count: typeof candidate.sample_count === 'number' ? candidate.sample_count : 0,
        success_count: typeof candidate.success_count === 'number' ? candidate.success_count : 0,
        failure_count: typeof candidate.failure_count === 'number' ? candidate.failure_count : 0,
        success_rate: typeof candidate.success_rate === 'number' ? candidate.success_rate : 0,
        empty_response_rate: typeof candidate.empty_response_rate === 'number' ? candidate.empty_response_rate : 0,
        rate_limit_count: typeof candidate.rate_limit_count === 'number' ? candidate.rate_limit_count : 0,
        avg_ttfb_ms: typeof candidate.avg_ttfb_ms === 'number' ? candidate.avg_ttfb_ms : 0,
        avg_total_ms: typeof candidate.avg_total_ms === 'number' ? candidate.avg_total_ms : 0,
        cooling_down: candidate.cooling_down === true,
        cooldown_remaining_ms: typeof candidate.cooldown_remaining_ms === 'number' ? candidate.cooldown_remaining_ms : 0,
        cooldown_reason: candidate.cooldown_reason ?? '',
        quota_status: candidate.quota_status ?? 'unknown',
        quota_balance: typeof candidate.quota_balance === 'number' ? candidate.quota_balance : undefined,
        quota_used: typeof candidate.quota_used === 'number' ? candidate.quota_used : undefined,
        effective_score: typeof candidate.effective_score === 'number' ? candidate.effective_score : 0,
        decision: candidate.decision ?? 'ready',
        notes: candidate.notes ?? [],
    };
}

function normalizePreview(preview: Partial<GroupRoutingPreview>): GroupRoutingPreview {
    return {
        group_id: typeof preview.group_id === 'number' ? preview.group_id : 0,
        group_name: preview.group_name ?? '',
        group_mode: typeof preview.group_mode === 'number' ? preview.group_mode : 1,
        health_score_enabled: preview.health_score_enabled === true,
        candidates: (preview.candidates ?? []).map(normalizeCandidate),
    };
}

export function useGroupRoutingPreview(groupId: number | undefined, enabled = true) {
    return useQuery({
        queryKey: ['group-routing', groupId],
        queryFn: async () => apiClient.get<GroupRoutingPreview>(`/api/v1/group/routing/${groupId}`),
        select: normalizePreview,
        enabled: enabled && typeof groupId === 'number' && groupId > 0,
        refetchInterval: enabled ? 30000 : false,
    });
}
